cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -197,17 +194,9 @@ func (csr *ClusterStateRegistry) RegisterOrUpdateScaleUp(nodeGroup cloudprovider
 	csr.registerOrUpdateScaleUpNoLock(nodeGroup, delta, currentTime)
 }
 
-// RegisterProviders registers providers in the cluster state registry.
-func (csr *ClusterStateRegistry) RegisterProviders(maxNodeProvisionTimeProvider maxNodeProvisionTimeProvider) {
-	csr.maxNodeProvisionTimeProvider = maxNodeProvisionTimeProvider
-}
-
 // MaxNodeProvisionTime returns MaxNodeProvisionTime value that should be used for the given NodeGroup.
 func (csr *ClusterStateRegistry) MaxNodeProvisionTime(nodeGroup cloudprovider.NodeGroup) (time.Duration, error) {
```

Comment:
I don't think we need this method either, it's body is literally just 1 line. Previously it was used to encapsulate the nil check, which is not relevant anymore IIUC?

Reply:
It's also called outside of this package, so this is needed if we want to keep the processor as a private value. I think it makes sense to leave it as is.

Reply:
Ok, I'd never expect that (because there's a dedicated processor with this functionality). The other callsites should definitely be using the processor for that and not CSR, this is a very roundabout way. Let's not bloat this PR further though, maybe add a TODO if you're up for it.

Reply:
Fully agree, added to do to sort this out in a follow up.